src/common.speech/DialogConnectorFactory.ts

```diff
@@ -66,18 +66,25 @@ export class DialogConnectionFactory extends ConnectionFactoryBase {
         const {resourcePath, version, authHeader} = getDialogSpecificValues(dialogType);
 
         const headers: IStringDictionary<string> = {};
-        headers[authInfo.headerName] = authInfo.token;
+        if (authInfo.token !== undefined && authInfo.token !== "") {
+            headers[authInfo.headerName] = authInfo.token;
+        }
         headers[QueryParameterNames.ConnectionIdHeader] = connectionId;
 
-        let endpoint: string;
-        // ApplicationId is only required for CustomCommands
-        if (applicationId === "") {
-            endpoint = `wss://${region}.${baseUrl}/${pathSuffix}/${version}`;
-        } else {
-            endpoint = `wss://${region}.${baseUrl}/${resourcePath}/${pathSuffix}/${version}`;
+        if (applicationId !== "") {
```

Comment:
same comment as above with null and undefined

Reply:
This one will have a value or be empty, per the call above where it's set